### PR TITLE
left align dashcard actions on small cards

### PR DESCRIPTION
**Pull request by npretto**
## Description

Redo of https://github.com/metabase/metabase/pull/35241/files
I talked with Dawei, he agreed on a easier solution with a fixed container query.
I based this off [#35324 fix z-issues on move to tab menu](https://github.com/metabase/metabase/pull/35324) to avoid conflicts in the props of the component.

In order to support container queries I had to update @emotion/* libraries, they're all minor updates, do we have a special procedure for updating libs?

## How to reproduce
1. Resize a card to the min size and put it on the first column
2. When hovering it in edit mode, the actions should be left aligned and not go out of the viewport

https://github.com/metabase/metabase/assets/1914270/26f1b633-051d-48ed-9027-05282b7c14ca

### CODE REVIEW

**Comment by npretto** [comment]
Current dependencies on/for this PR:
* master
  * **PR #35240** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35240" target="_blank">Graphite</a> 
    * **PR #35324** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35324" target="_blank">Graphite</a> 
      * **PR #35327** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35327" target="_blank">Graphite</a>  👈
    * **PR #35309** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35309" target="_blank">Graphite</a> 
      * **PR #35310** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35310" target="_blank">Graphite</a> 
    * **PR #35243** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35243" target="_blank">Graphite</a> 
    * **PR #35241** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35241" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/35327?utm_source=stack-comment).

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx [hunk 1]
```diff
@@ -57,6 +61,12 @@ export const DashboardCardActionsPanel = styled.div<{ isSubMenuOpen: boolean }>`
   // react-resizable covers panel, we have to override it
   z-index: 2;
 
+  // left align on small cards on the left edge to not make the actions go out of the viewport
+  @container (max-width: 200px) {
```
Comment (npretto) [inline frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx:65]: 200px makes it a good number for cards with 5 buttons, which AFAIK is the max we can have at the moment

https://github.com/metabase/metabase/assets/1914270/b8b3b8a1-dfed-459c-a7ba-f89176c6e750

Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx:65]: (red color is just to show when the container query applies, same for opacity always at 1 on the actions panel)

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx [hunk 2]
```diff
@@ -57,6 +61,12 @@ export const DashboardCardActionsPanel = styled.div<{ isSubMenuOpen: boolean }>`
...
+    ${({ onLeftEdge }) => onLeftEdge && "right: unset;"}
+    ${({ onLeftEdge }) => onLeftEdge && "left: 20px;"}
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx:67]: We usually use rem for sizes

**Review by ranquild — APPROVED** [review]
Please take a look at the comment

**Comment by npretto** [comment]
Merging this into the other PR to have less conflicts in another PR that I'll open soon

**Pull request merged**